src/taco-utils/telemetry.ts

```diff
@@ -225,23 +227,25 @@ module TacoUtility {
                     context.tags[context.keys.applicationVersion] = appVersion;
                 }
 
-                TelemetryUtils.userId = TelemetryUtils.getUserId();
-                TelemetryUtils.machineId = TelemetryUtils.getMachineId();
-                TelemetryUtils.sessionId = TelemetryUtils.generateGuid();
-                TelemetryUtils.userType = TelemetryUtils.getUserType();
-                
-
-                if (_.isUndefined(isOptedInValue)) {
-                    return TelemetryUtils.getOptIn()
-                    .then(function (optIn: boolean): void {
-                        Telemetry.isOptedIn = optIn;
+                return Q.all([TelemetryUtils.getUserId(), TelemetryUtils.getMachineId()])
```

Comment:
We have been writing {"userId":"X-X","machineId":"Y-Y","userType":"Z","optIn":false} even if the user opts out. I do see that it seems a bit unnecessary. What do you suggest? We track the current behavior as an issue if we decide to.